# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/uif2iso/gcc15.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Fix K&R forward declarations for C23 (gcc-15+).

Copyright (C) 2026 René Rebe <rene@exactco.de>
Signed-off-by: René Rebe <rene@exactco.de>

--- a/loki91.c	2009-01-06 15:21:12.000000000 +0100
+++ b/loki91.c	2026-09-25 19:30:28.231531143 +0200
@@ -139,8 +139,8 @@ short	gen;		/* irreducible polynomial ge
 char	*loki_lib_ver = "LOKI91 library v3.0, Copyright (C) 1991 Lawrence Brown & UNSW";
 
 		/* subkeys at the 16 rounds */
-static Long	f();			/* declare LOKI function f */
-static short	s();			/* declare LOKI S-box fn s */
+static Long	f(Long, Long);		/* declare LOKI function f */
+static short	s(Long);		/* declare LOKI S-box fn s */
 
 
 /*
@@ -302,7 +302,6 @@ static short s(i)
 register Long i;	/* return S-box value for input i */
 {
 	register short	r, c, v, t;
-	short	exp8();		/* exponentiation routine for GF(2^8) */
 
 	r = ((i>>8) & 0xc) | (i & 0x3);		/* row value-top 2 & bottom 2 */
 	c = (i>>2) & 0xff;			/* column value-middle 8 bits */
